pkg/artifactory/datasource/repository/federated/federated.go

```diff
@@ -0,0 +1,36 @@
+package federated
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+const rclass = "federated"
+
+var MemberSchema = map[string]*schema.Schema{
```

Comment:
It was redefined here because the MemberSchema is Optional and not Required. 

The data source will always return with a member schema, but a member schema isn't required to retrieve the repository, hence it needs to redefined. (At least thats what I thought)

Reply:
Ah yes. I ran into similar issue with remote's `url` attribute. Instead of redefining everything again, multiple times, I created a generator function instead.

https://github.com/jfrog/terraform-provider-artifactory/blob/master/pkg/artifactory/resource/repository/remote/remote.go#L101

Reply:
Great idea, will do that